samples/caf_sensor_manager/src/modules/sensor_sim_ctrl.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <drivers/sensor_sim.h>
+
+#include "sensor_sim_ctrl_def.h"
+
+#define MODULE sensor_sim_ctrl
+#include <caf/events/module_state_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE);
+
+enum state {
+	STATE_DISABLED,
+	STATE_ACTIVE,
+	STATE_ERROR
+};
+
+static enum state state;
+static uint8_t cur_wave_idx;
+static const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(sensor_sim));
+
+static void report_error(void)
+{
+	state = STATE_ERROR;
+	module_set_state(MODULE_STATE_ERROR);
+}
+
+static void set_wave(void)
+{
+	const struct sim_wave *w = &sim_signal_params.waves[cur_wave_idx];
+	int err = sensor_sim_set_wave_param(dev, sim_signal_params.chan, &w->wave_param);
+
+	if (err) {
+		LOG_ERR("Cannot set simulated accel params (err %d)", err);
+		report_error();
+	}
+}
+
+static void verify_wave_params(void)
+{
+	__ASSERT_NO_MSG(sim_signal_params.waves != NULL);
+	__ASSERT_NO_MSG(sim_signal_params.waves_cnt > 0);
+
+	for (size_t i = 0; i < sim_signal_params.waves_cnt; i++) {
+		const struct sim_wave *w = &sim_signal_params.waves[i];
+
+		__ASSERT_NO_MSG(w->label != NULL);
+		__ASSERT_NO_MSG(w->wave_param.type < WAVE_GEN_TYPE_COUNT);
+		__ASSERT_NO_MSG((w->wave_param.period_ms > 0) ||
+				(w->wave_param.type == WAVE_GEN_TYPE_NONE));
+		ARG_UNUSED(w);
+	}
+}
+
+static int init(void)
+{
+	if (IS_ENABLED(CONFIG_ASSERT)) {
+		verify_wave_params();
+	}
+
+	set_wave();
+
+	return 0;
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_module_state_event(eh)) {
+		const struct module_state_event *event = cast_module_state_event(eh);
+
+		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+			__ASSERT_NO_MSG(state == STATE_DISABLED);
+
+			if (!init()) {
+				module_set_state(MODULE_STATE_READY);
+				state = STATE_ACTIVE;
+			} else {
+				report_error();
+			}
```

Comment:
Currently this is pointless but if you add `static int init()` it will make sense to keep this.